proposals/0002-cxx-attributes.md

```diff
@@ -124,4 +124,65 @@ Below are a few more examples of C++ attributes that we could support:
     }
   }
 ```
+## Detailed design
+
+The simplest explanation of this feature is supporting C++11 attribute syntax on
+all shared grammar elements between HLSL and C++. This spec attempts to detail
+some of the grammar and parsing implications and will specify the process by
+which existing attributes will convert to C++ attributes.
+### Attribute Parsing

```

Comment:
Best to add a blank line before the section header for the benefit of those reading the markdown directly (I don't think it makes a difference in the rendered doc)